### ISSUE TITLE
Issue while using clingo with asprin

### ISSUE THREAD

**Issue by PaoloPangallo** [potassco/asprin#13]
I have clingo 5.7.1 and asprin 3.1.0 and it seems that I can t run asprin proprerly 
![image](https://github.com/user-attachments/assets/f8ee8da5-b074-4d85-ad4e-061c8ec969b8)

the same error shows up when i try to run the followingcode on pycharm :

import sys
print("Python version:")
print(sys.version)

try:
    import clingo
    print("Clingo imported successfully")
    print("Clingo version:")
    print(clingo.version)
except ImportError as e:
    print("Failed to import clingo:", e)

import subprocess

def run_asprin():
    try:
        result = [subprocess.run](https://subprocess.run/)(['asprin', '--version'], capture_output=True, text=True, check=True)
        print("Asprin ok, output:")
        print(result.stdout)
    except subprocess.CalledProcessError as e:
        print("Error:", e)
        print("stdout:", e.stdout)
        print("stderr:", e.stderr)
    except Exception as e:
        print("Unexpected error", e)

#
run_asprin()

![image](https://github.com/user-attachments/assets/f4e394e7-df67-447f-81b3-9c6a76db1fe6)

**Comment by rkaminsk** [comment]
I don't think this has anything to do with clingo. It might be that you have to use an older version. I'll transfer the issue to the asprin repo.

**Comment by tortinator** [comment]
Yes @PaoloPangallo , you are right! We are just restarting to refactor asprin, but AFAIK it works fine in the settings given in the README:

asprin has been tested with Python 2.7.13 and 3.5.3, using clingo 5.4.0.

**Comment by PaoloPangallo** [comment]
I have issues installing your configuration because clingo 5.4.0 doesnt work with the python version given, what do you suggest?

![image](https://github.com/user-attachments/assets/994602c8-ef39-4ea6-a0a0-285ce899a678)

**Comment by rkaminsk** [comment]
> I have issues installing your configuration because clingo 5.4.0 doesnt work with the python version given, what do you suggest?

I would use at least python 3.8. Anything before is end of life. Most likely asprin will work with that too.

**Comment by tortinator** [comment]
I'm afraid we need to wait for this until @javier-romero is. Ack from vacation

**Comment by javier-romero** [comment]
Hi @PaoloPangallo, 

I am not sure about what is the problem in your case. 

In my machine it works well with the versions of the README. 
For example, using conda:
```
conda create -n asprin -c conda-forge -c potassco python=3.5.3 clingo=5.4.0 asprin
```

These other installations (with other Python versions) should also work:
```
conda create -c potassco -n asprin python=3.7 clingo=5.4 asprin=3.1
conda create --strict-channel-priority -c conda-forge -c potassco -n asprin python=3.9 clingo=5.4 asprin=3.1
```

**Comment by mphizi** [comment]
@javier-romero actually there are 2 issues I found so far with clingo 5.7+. Imports handling changed with change in python version - this creates the startup "clingo does not have a module called ast" or something like that. I fixed that by just importing "clingo.ast" after importing clingo. 

The second issue I still can't solve: in the old clingo one retrieved a ProgramBuilder object with Control.builder() but this method no longer exists on the Control object of clingo so there's an error because asprin calls this method in program_parser.py (lines 427 and 433). Using asprin v3.1.1.

**Comment by mphizi** [comment]
> [@javier-romero](https://github.com/javier-romero) actually there are 2 issues I found so far with clingo 5.7+. Imports handling changed with change in python version - this creates the startup "clingo does not have a module called ast" or something like that. I fixed that by just importing "clingo.ast" after importing clingo.
> 
> The second issue I still can't solve: in the old clingo one retrieved a ProgramBuilder object with Control.builder() but this method no longer exists on the Control object of clingo so there's an error because asprin calls this method in program_parser.py (lines 427 and 433). Using asprin v3.1.1.

Actually it seems that the version deployed to anaconda does not have the clingo 5.7 compatibility fixes. It might help to have a new release with those fixes. 

For those interested, just replace the files in `/site-packges/asprin/` with the file in the asprin directory on the `Master` branch at least as of the date of this comment.

Kudos @halex97 for the fixes!!
